src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/AttributePresenceFilterNode.cs

```diff
@@ -0,0 +1,295 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using Internal.JitInterface;
+using Internal.NativeFormat;
+using Internal.Text;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    public class AttributePresenceFilterNode : HeaderTableNode
+    {
+        private EcmaModule _module;
+        private ObjectData computedData;
+
+        public override int ClassCode => 56456113;
+
+        public AttributePresenceFilterNode(EcmaModule module, TargetDetails target)
```

Comment:
Nit: `TargetDetails` can be extracted from `module.Context.Target`

Reply:
Fixed in next iteration

Reply:
Fixed